jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -72,7 +72,7 @@ public Tracer getTracer() {
     public Endpoint getPeer() {
         synchronized (this) {
             if (peer == null) {
-                peer = new Endpoint();
+                peer = new Endpoint(0, (short) 0, "");
```

Comment:
might want to return null.. ex null endpoint is ok, it is null serviceName that's the problem. an empty endpoint w/o an ip or serviceName or port, is probably worse than null.